src/main/prepared_statement_data.cpp

```diff
@@ -48,4 +49,36 @@ LogicalType PreparedStatementData::GetType(idx_t param_idx) {
 	return it->second[0]->type();
 }
 
+void PreparedStatementData::ResetResultValueEntry(idx_t val_idx, const Value &value) {
+	D_ASSERT(val_idx > 0);
+	auto it = value_map.find(val_idx);
+	if (it == value_map.end()) {
+		throw BinderException("Could not find parameter with index %llu", val_idx);
```

Comment:
Same here - exceptions are never allowed in the ODBC client. Exceptions crossing the C-API boundary is undefined behavior.